packages/sdk/src/utils/getAssetByChain.ts

```diff
@@ -0,0 +1,23 @@
+import type { Asset, AssetData, AssetFuel } from '@fuel-wallet/types';
+
+export function getAssetByChain(
+  asset: Asset,
+  chainId: number,
+  network: string = 'fuel'
+): AssetData {
+  const assetFuelNetwork = asset.networks.find(
+    (item) => item.chainId === chainId && item.type === network
+  ) as AssetFuel;
+
+  if (!assetFuelNetwork) {
+    throw new Error('Asset not found for the given chain and network.');
+  }
+
+  return {
+    ...asset,
+    assetId: assetFuelNetwork.assetId,
+    decimals: assetFuelNetwork.decimals,
+    chainId: assetFuelNetwork.chainId,
+    network: assetFuelNetwork.type,
+  };
+}
```

Comment:
instead of having this, we can use from `assets` new utils function ?

Reply:
We can do this in the following PR